codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostRequestEncoder.java

```diff
@@ -206,9 +215,8 @@ public HttpPostRequestEncoder(
             throw new NullPointerException("charset");
         }
         HttpMethod method = request.method();
-        if (!(method.equals(HttpMethod.POST) || method.equals(HttpMethod.PUT)
-                || method.equals(HttpMethod.PATCH) || method.equals(HttpMethod.OPTIONS))) {
-            throw new ErrorDataEncoderException("Cannot create a Encoder if not a POST");
+        if (method.equals(HttpMethod.TRACE)) {
+            throw new ErrorDataEncoderException("Cannot create a Encoder if request is a TRACE");
```

Comment:
For the moment, I let it like that, since I feel, maybe wrongly, that changing the signature / behaviour here (IllegalArgumentException instead of ErrorDataEncoerException as originally) may lead to incorrect behavior in end-user code if they were depending on this exception, as noted in the Javadoc. It seems part of the contract. And so changing it will lead to change the API contract.

I'm ready to change it however, but @Scottmitch WDYT ?